apis/python/tests/test_dataframe.py

```diff
@@ -783,7 +783,9 @@ def make_multiply_indexed_dataframe(
             "domain": [[-1000, 1000]],
             "coords": [{"bogus": True}],
             "A": None,
-            "throws": TypeError,
+            # Disable Typeguard while asserting this error, otherwise a typeguard.TypeCheckError is
```

Comment:
This change isn't due to using Typeguard@HEAD; it will remain relevant when we move back to a proper Typeguard release.

The reason for it is: in 3.12, better type-checking was causing a `TypeCheckError` to occur before the `TypeError` we were verifying here.

I worked around it by disabling type-checking while triggering this error, so we can continue to verify what we were previously.